mainloop.cpp

```diff
@@ -59,6 +59,8 @@ int Mainloop::open()
         return -1;
     }
 
+    Endpoint::set_Mainloop(this);
```

Comment:
I'd prefer to be the other way around. There shouldn't be a "set_Mainloop()" method.

Rather add a static method `Mainloop::get_instance()`. Whoever needs it can call this method that will return the only Mainloop instance.